src/Share/Postgres/Hashes/Queries.hs

```diff
@@ -307,6 +305,31 @@ expectNamespaceIdsByCausalIdsOf trav s = do
         then unrecoverableError . MissingExpectedEntity $ "expectNamespaceIdsByCausalIdsOf: Expected to get the same number of results as causal ids. " <> tShow causalIds
         else pure results
 
+-- | Mitchell says: this could/should just have replaced 'expectNamespaceIdsByCausalIdsOf', but that function has
+-- many callers, so having two temporarily eases the transition.
```

Comment:
Hrmm, I wonder maybe the better option might be to absorb `pUnrecoverableEitherMap` into the `QueryA` class, that way you can just have a `QueryA m` constraint here and wouldn't need to touch any callsites.

I admit, the transaction types are getting away from me a bit as the number of them slowly grows 😓